UserListScreen.tsx

```diff
@@ -1,15 +1,135 @@
 import React from 'react'
 import { Component } from 'react';
-import { Text } from 'react-native';
+import { Text, View, Image, FlatList, StyleSheet, ActivityIndicator, AppRegistry } from 'react-native';
+import { ListItem } from 'react-native-elements';
+
+import { ApolloClient } from 'apollo-client';
+import { ApolloProvider, Query } from 'react-apollo';
+import { InMemoryCache } from 'apollo-cache-inmemory';
+import { createHttpLink } from 'apollo-link-http';
+import { setContext } from 'apollo-link-context';
+import gql from "graphql-tag";
+
+import ApolloApp from './App';
+
+import { AsyncStorage } from 'react-native';
+
+const httpLink = createHttpLink({
+  uri: "https://tq-template-server-sample.herokuapp.com/graphql"
+});
+
+const authLink = setContext(async (_, { headers }) => {
+  const token = await AsyncStorage.getItem('@Token:key');
+  return {
+    headers: {
+      ...headers,
+      authorization: token ? `${token}` : "",
```

Comment:
1. O token já não é uma string? Fazer ``${token}`` fica redundante.
2. Será que enviar uma string vazia nesse header é a melhor coisa a se fazer caso o token não exista? 🤔

Reply:
Eu já estava tratando no `error` da Query essa exceção e não percebi.
Deixei `authorization: token` no código.

Reply:
Não precisa alterar isso, mas fica a reflexão: no caso de não ter token, será que vale a pena fazer a request, mesmo sabendo q ela é autenticada, e não tem um token no armazenamento?

Reply:
Entendi, eu poderia pegar o token assim que entra no `render()` e se for vazio nem envia a requisição.